lesson_7/dobricsongor/decorator.py

```diff
@@ -0,0 +1,65 @@
+# Create a decorator called uppercase that will uppercase the result
+
+def uppercase(fnc):
+    
+    def inner_func(given_text):
+        return(fnc(given_text).upper())
+    
+    return inner_func
+    
+
+@uppercase
+def greet(name):
+    return "Greetings {}!".format(name)
+
+print(greet("World"))
+
+
+#------------------------------------------------------------------------------
+
+#Create a decorator called safe_divide that will output a message if the division cannot be performed, othervise it will return the result.
+
+def safe_divide(fnc):
+
+    def inner_func(first_num,second_num):
+        if isinstance(first_num, (int,float)) and isinstance(second_num, (int,float)): 
+            #if first_num % second_num == 0:
+            print(f'The result of dividing the numbers {first_num} and {second_num} is: ',fnc(first_num,second_num))
+        else:
+            print(f'The operation divide of {first_num} and {second_num} cannot be done!') 
+
+    return inner_func
+
+@safe_divide
+def divide(first_number, second_number):
+    return first_number / second_number
+
+divide(10.3, 2)
```

Comment:
What would happen if we will try the following division: 3/0 ?

Reply:
Ma gandeam ca division 3/0 se poate face, doar ca rezultatul e 0. 
Rezolv :)